modules/agent-configuration-parameters.adoc

```diff
@@ -0,0 +1,122 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="agent-configuration-parameters_{context}"]
+= Available Agent configuration parameters
+The following tables specify the required and optional Agent configuration parameters that you can set as part of the Agent-based installation process.
+
+These values are specified in the `agent-config.yaml` file.
+
+[NOTE]
+====
+These settings are used for installation only, and cannot be modified after installation.
+====
+
+[id="agent-configuration-parameters-required_{context}"]
+== Required configuration parameters
+
+Required Agent configuration parameters are described in the following table:
+
+.Required parameters
+[cols=".^2l,.^3,.^5a",options="header"]
+|====
+|Parameter|Description|Values
+
+|apiVersion:
+|The API version for the `agent-config.yaml` content.
+The current version is `v1beta1`.
+The installation program may also support older API versions.
+|String
+
+|metadata:
+|Kubernetes resource `ObjectMeta`, from which only the `name` parameter is consumed.
+|Object
+
+|metadata:
+  name:
+|The name of the cluster.
```

Comment:
The parameter `metadata.name` is optional; however, when included in the agent-config file, it takes precedence over the value specified in the install-config file.

Reply:
If included in the agent-config file, the metadata.name parameter neither takes precedence nor serves as a required or optional setting; rather, it is simply ignored, and the value specified in the install-config file is used.
@bfournie please confirm.

Reply:
The metadata name and namespace are required. From the agent-config.yaml defintion: https://github.com/openshift/installer/blob/master/pkg/types/agent/agent_config_type.go#L19-L20

However they are not validated or used. As Manoj indicated, the fields are ignored

Reply:
I'll add a sentence to this entry stating "The value entered in the `agent-config.yaml` file is ignored, and the instead the value specified in the `install-config.yaml` file is used."

Let me know if that sounds right, or if I need to add to/change that sentence.

Reply:
That sounds right, although I did discover that if the install-config.yaml is ALSO not supplied that a default cluster name of `agent-cluster` will be used, see https://github.com/openshift/installer/blob/master/pkg/asset/agent/installconfig.go#L253-L258

Reply:
@bfournie just out of curiosity, how is this value a required parameter in both the agent-config and install-config files if the installation proceeds without them specified? I'm not familiar with OCP configuration, does "required" not mean "necessary for the installation and thus the installer will terminate with an error without this value"?

Reply:
@bfournie If the value is not specified in the install-config, the following error message will be generated: "FATAL failed to fetch Agent Manifests: failed to load asset "Install Config": invalid install-config configuration: metadata.name: Invalid value: "": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character"

Reply:
For the install-config.yaml, if it's provided the metadata.name must be included. However install-config.yaml is optional for the agent installer (when using just the ZTP manifests). In that case the cluster name will be set to `agent-cluster`

Reply:
Proposed sentence to add to `metadata.name` descriptions for both agent-config and install-config: 

"When you do not provide `metadata.name` through either the `install-config.yaml` or `agent-config.yaml` files, for example when you use only ZTP manifests, the cluster name will be set to `agent-cluster`."